### PR TITLE
Upgraded code base to support PHP 8.0 & PHPUnit 9

### PR DESCRIPTION
**Pull request by carlos-mg89**
I've created a new PR, since I'm going to make more changes in my personal fork at the master branch. Plus, I've added PHP 7.3 again.

I need to make it PHP 8 only, due to `psr/cache` requiring it, and I need to use Symfony 6.

So this PR replaces https://github.com/samwilson/phpflickr/pull/46

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,20 +14,20 @@
...
 	},
 	"require": {
-		"php": ">=5.6",
+		"php": "^7.3 || ^8.0",
 		"ext-curl": "*",
 		"ext-json": "*",
 		"ext-libxml": "*",
 		"ext-simplexml": "*",
-		"lusitanian/oauth": "^0.8",
+		"lusitanian/oauth": "dev-master#ee5a83310c6014b6cc07ac0610ec9d67ba452664 as 0.8.12",
```
Comment (samwilson) [inline composer.json:22]: I do wonder if we should switch to your fork of this.

**Review by samwilson — COMMENTED** [review]
I'm really sorry for being so slow to get to this!

Merging now.

File: composer.json (hunk 1 above)
Comment (carlos-mg89) [reply composer.json:22]: That's totally up to you. I'd be very honored, and I'd do my best to keep it up-to-date. I don't intend to leave my fork unattended. I'm using it in several personal and professional projects.

https://github.com/carlos-mg89/PHPoAuthLib/blob/master/composer.json

The biggest difference is that my fork allows PHP ^8.0 and the lusitanian only PHP 8.0.

Comment (samwilson) [reply composer.json:22]: It doesn't sound like there's going to be much action on the lusitanian/oauth front, so it seems worth making a new fork.

I want to add PHP 8.1 support too (which I don't think requires much).

Comment (carlos-mg89) [reply composer.json:22]: The fork I created, is compatible so far with PHP 8.1 too. Actually, that is the main reason I forked lusitanian/oauth project, to make it compatible with PHP 8.0 and PHP 8.1.

**Pull request merged**